gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/TagsValidationServiceImpl.java

```diff
@@ -72,4 +81,18 @@ public Set<String> validateAndSanitize(final ExecutionContext executionContext,
 
         return newTags;
     }
+
+    public void validatePlanTagsAgainstApiTags(Set<String> planTags, Api api) {
```

Comment:
To avoid having V2/V4 distinction in this service, do you think we could directly give API tags in the second parameter instead of the "API" itself? Doing that, you would move the specific code to the dedicated plan service? WDYT?

Reply:
I did it first but in that case we do not mutualize the JSON parsing and so I was thinking that the interest of the `validatePlanTagsAgainstApiTags` was limited.